src/arch/riscv/kernel/context.c

```diff
@@ -1,11 +1,15 @@
-#include <kernel/panic.h>
+#include <stdint.h>
 #include <hal/context.h>
+#include <hal/ipl.h>
+#include <asm/regs.h>
 
 void context_init(struct context *ctx, unsigned int flags,
 		void (*routine_fn)(void), void *sp) {
+	ctx->sp = (uint32_t) sp;
```

Comment:
`ctx->sp` is `unsigned int`, but you cast `sp` to `(uint32_t)`, I recommend to use `uint32_t` explicitly for `ctx` fields (if they are 32 bit wide).